### ISSUE TITLE
Solved: Kernel Panic DVMT

### ISSUE THREAD

**Issue by igormclven** [jsassu20/Lenovo-ThinkPad-T450-Mojave#3]
Hello, 

To use your configuration, do I have to make any special settings ? I replaced my EFI partition with your 1.2 release but I get kernel panic. I'm not sure it can be, I got the following:

(I'm sorry about the quality, the error happens very fast and I almost didn't get it on video.)
![Screenshot_2019-05-14-20-32-26-489_com google android apps photos](https://user-images.githubusercontent.com/8197834/57743639-88d98800-768b-11e9-9365-6148a1b584a8.png)

**Comment by jsassu20** [comment]
Ok I just updated everything again about 3 minutes ago. Try out the new setup and if the first config doesn't work try the config_alt.plist by going into options at the clover screen and selecting the config option.

**Comment by igormclven** [comment]
Thank you very much, I'll try. :)

**Comment by igormclven** [comment]
No, in all three config I get kernel panic.
The only difference between your setup and mine is that I have 8gb of RAM.  The dedicated memory is at 512 mb. :c

**Comment by jsassu20** [comment]
Take a picture of you bios pages and text them to me 8605180536 it’s iMessage if you have an iPhone or you can just use that. Don’t worry buddy I’m gonna get your system running. I won’t leave you hanging. I just gotta know some shit about your setup

**Comment by svobluk** [comment]
Hello,

unfortunately I have exact the same issue like igormclven has. Kernel Panic at "Panic at FB memory manager" I have tried all three configs as well, but no lock. Also tried 256 or 512 mb for memory allocation. Can't find any differences which point me to the kernel panic root case now.

I have 5300U and 16GB RAM.

**Comment by svobluk** [comment]
> Hello,
> 
> unfortunately I have exact the same issue like igormclven has. Kernel Panic at "Panic at FB memory manager" I have tried all three configs as well, but no lock. Also tried 256 or 512 mb for memory allocation. Can't find any differences which point me to the kernel panic root case now.
> 
> I have 5300U and 16GB RAM.

I have an answer to myself :-)

I have found a solution; I have added framebuffer-stolenmem and framebuffer-fbmem patch. It has solved the KP inside the system and for recovery partition as well. (I'll have a look how to "unlock" BIOS for bigger DVMT anyway, but it works now with default 32MB!

**Comment by jsassu20** [comment]
The issue is a framebuffer malfunction related to the presence of the IMEI device or the configured memory in your BIOS for the video display for DMVT. My BIOS is modified so I can change mine at will, other systems are unfortunately fixed at a specific amount of memory based on a few different factors. I think its most likely that yours was set to 32MB and the IGPU configuration I was using was causing the kernel panic because my configuration utilizes 128MB DVMT. Ive updated the release with the 32MB DVMT IGPU configuration settings in ACPI/Devices/Properties/PciRoot(0x0)/Pci(0x2,0x0).... See if this doesn't take care of your issue. I'm going to create a few different config.plists for different circumstances and label them for easy use. Give me about 30 minutes and I'll post the new build.

**Comment by jsassu20** [comment]
Oh wow lol I didn't see your message sorry about that Im glad you managed to work it out on your own

**Comment by igormclven** [comment]
Of course! I had read about this some time ago but I didn't follow up on it precisely because the BIOS had to be patched. Thank you very much for the new files. Although I would like to know how to patch the bios in order to make those changes, some time ago I read that you had to make some adjustments to the hardware and use an electronic controller and several other things to get a patched BIOS, but I don't know if it's any other way now. 

Thank you.

**Comment by jsassu20** [comment]
Its pretty simple. If you purchase a USB Programmer CH341A Series 25 SPI Flash Tool and Clip that connects to it I can walk you through the process of dumping your bios into the proper BIN file needed then you can upload it to me and I will modify your BIOS with the proper changes using a HEX tool which would disable all whitelists present (including the LTE card and the Display which fixes the ability to control brightness under windows when you add a 1080p display to the T450) and unlock the Advanced menu in the BIOS which gives you complete control over the the device. It is a bit dangerous though having that much control over the device so be careful not to adjust things you aren't familiar with but if you just ask me first I'd be happy to tell you what not to touch. You can get both items on amazon for under 20 dollars together...

https://www.amazon.com/OctagonStar-SOIC8-Socket-Adpter-Programmer/dp/B01M8G1GJ2/ref=pd_bxgy_328_img_3/133-3623101-8957635?_encoding=UTF8&pd_rd_i=B01M8G1GJ2&pd_rd_r=d3f06982-79aa-11e9-b98b-a55320510026&pd_rd_w=d4bgg&pd_rd_wg=Ibzah&pf_rd_p=a2006322-0bc0-4db9-a08e-d168c18ce6f0&pf_rd_r=6X13SACWJDR3M7XJWSWB&psc=1&refRID=6X13SACWJDR3M7XJWSWB

https://www.amazon.com/Gikfun-Programmer-CH341A-Burner-EEPROM/dp/B01I1EU9LG/ref=pd_bxgy_328_img_2/133-3623101-8957635?_encoding=UTF8&pd_rd_i=B01I1EU9LG&pd_rd_r=d3f06982-79aa-11e9-b98b-a55320510026&pd_rd_w=d4bgg&pd_rd_wg=Ibzah&pf_rd_p=a2006322-0bc0-4db9-a08e-d168c18ce6f0&pf_rd_r=6X13SACWJDR3M7XJWSWB&psc=1&refRID=6X13SACWJDR3M7XJWSWB

**Comment by igormclven** [comment]
Thank you for your work. I will order the parts to get the most out of this equipment. I'll be very grateful for your help later with that. 

I'll let you know soon. Thank You.

**Comment by igormclven** [comment]
Hey, dude, how you doing?

Amazon purchases have arrived with the Ch341 programmer. Could you please help me with the following process. Thank you very much. You're the best, dude.
